### ISSUE TITLE
[Feature Request] Make tmdb and tvmaze pass through global proxy too

### ISSUE THREAD

**Issue by papampi** [pymedusa/Medusa#7546]
Adding new and existing shows when tmdb and tvmaze indexers can not be reached get long time and almost impossible to start fresh.
I just migrated from sickchill to medusa, tried to start clean and in 2 hours it only added 4 existing shows with these errors:
```
2020-01-04 19:38:02 WARNING  SHOWQUEUE-ADD :: [d0c136d] Error loading IMDb info: HTTPSConnectionPool(host='api.imdbws.com', port=443): Max retries exceeded with url: /authentication/credentials/temporary/ios82 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55f41c48e0>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:33:35 WARNING  SHOWQUEUE-ADD :: [d0c136d] Could not get tvmaze externals. Cause: ConnectTimeout(MaxRetryError("HTTPConnectionPool(host='api.tvmaze.com', port=80): Max retries exceeded with url: /lookup/shows?imdb=tt4941240 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f55f408b360>, 'Connection to api.tvmaze.com timed out. (connect timeout=30)'))",),)
2020-01-04 19:30:31 WARNING  SHOWQUEUE-ADD :: [d0c136d] Could not get tvmaze externals. Cause: ConnectTimeout(MaxRetryError("HTTPConnectionPool(host='api.tvmaze.com', port=80): Max retries exceeded with url: /lookup/shows?thetvdb=311810 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f55f404eed0>, 'Connection to api.tvmaze.com timed out. (connect timeout=30)'))",),)
2020-01-04 19:25:14 WARNING  SHOWQUEUE-ADD :: [d0c136d] Could not get tvmaze externals. Cause: ConnectTimeout(MaxRetryError("HTTPConnectionPool(host='api.tvmaze.com', port=80): Max retries exceeded with url: /lookup/shows?imdb=tt4941240 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f55f401a8f0>, 'Connection to api.tvmaze.com timed out. (connect timeout=30)'))",),)
2020-01-04 19:22:03 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec0f9930>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:19:52 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec0025a0>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:17:41 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec00be50>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:15:30 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec18ff60>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:13:19 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec0d4490>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:11:08 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec1fec00>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:08:57 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec029190>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-04 19:06:46 WARNING  SHOWQUEUE-ADD :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55ec0f0cd0>: Failed to establish a new connection: [Errno 110] Connection timed out',))
```

If passing tmdb and tvmaze through proxy too when "Use proxy for indexers" is enabled it would solve the issue.

**Comment by papampi** [comment]
Main issue posted here :https://github.com/pymedusa/Medusa/issues/7545

**Comment by medariox** [comment]
This is doable for TVDBv2 and TVmaze, but not for TMDB as the library we use doesn't use custom requests sessions.

**Comment by papampi** [comment]
At least one step ahead.
Possible to lower retry limits or session timeouts or disable tmdb if for example it gives error 5 consecutive times or something similar then retry again later in next update?
Or even add an option in settings to completely disable it when user know it gonna gives timeout error.

When I check log files, it takes 2-3 minutes times before it gives error:
```
2020-01-05 00:37:47 WARNING	SHOWQUEUE-UPDATE :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55dc0cc950>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-05 00:35:38 DEBUG	SHOWQUEUE-UPDATE :: [d0c136d] User-Agent: Medusa/0.3.9 (Linux; 4.14.24-qnap; 730f3651-2efd-11ea-827f-00089bef61c6)
```

```
2020-01-05 00:24:53 WARNING	SHOWQUEUE-UPDATE :: [d0c136d] Indexer TMDB is unavailable at this time: HTTPSConnectionPool(host='api.themoviedb.org', port=443): Max retries exceeded with url: /3/configuration?api_key=edc5f123313769de83a71e157758030b (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f55d823ea80>: Failed to establish a new connection: [Errno 110] Connection timed out',))
2020-01-05 00:22:44 DEBUG	SHOWQUEUE-UPDATE :: [d0c136d] Searching for fanart for series 364874
```

**Comment by papampi** [comment]
Strange, I just disabled proxy and connected to "PIA" openvpn from my Qnap where Medusa is installed and Medusa still giving the errors and cant connect to tvmaze
I can request api with curl with no problems...

My show update from 2 nights ago hasn't finished yet and still so many updates in server status :(

**Comment by papampi** [comment]
Its Weird...
I setup my PIA vpn on my router and all my lan devices are connected to internet through vpn.
Removed proxy from Medusa config but I still get the errors for tmdb and imdb.
I can open them with no problems, any idea?

**Comment by p0psicles** [comment]
This has been implemented in v 0.5.0
